acme/http.go

```diff
@@ -33,9 +34,11 @@ func httpHead(url string) (resp *http.Response, err error) {
 	req.Header.Set("User-Agent", userAgent())
 
 	resp, err = http.DefaultClient.Do(req)
-	if resp.Body != nil {
-		resp.Body.Close()
+	if err != nil {
+		return resp, err
 	}
+	io.Copy(ioutil.Discard, resp.Body)
```

Comment:
I agree with the bug fix, but not with the io.Copy, as far as my understanding of net/http goes.